bundles/org.openhab.binding.renault/src/main/java/org/openhab/binding/renault/internal/RenaultHandler.java

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.renault.internal;
+
+import static org.openhab.binding.renault.internal.RenaultBindingConstants.*;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.renault.internal.renault.api.Car;
+import org.openhab.binding.renault.internal.renault.api.MyRenaultHttpSession;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.PointType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link RenaultHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Doug Culnane - Initial contribution
+ */
+@NonNullByDefault
+public class RenaultHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(RenaultHandler.class);
+
+    private @Nullable RenaultConfiguration config;
+
+    private @Nullable ScheduledFuture<?> pollingJob;
+
+    public RenaultHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // This binding only polls status data automatically.
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.UNKNOWN);
+        this.config = getConfigAs(RenaultConfiguration.class);
+
+        // Background initialization:
+        if (pollingJob == null || pollingJob.isCancelled()) {
+            pollingJob = scheduler.scheduleWithFixedDelay(this::getStatus, 0, config.refreshInterval, TimeUnit.MINUTES);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        super.dispose();
+    }
+
+    private void getStatus() {
+        MyRenaultHttpSession httpSession;
+        try {
+            httpSession = new MyRenaultHttpSession(this.config);
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
```

Comment:
Just `updateStatus(ThingStatus.ONLINE);`
In case httpSession.updateCarData can trigger an exception, I will move this line just after.

Reply:
Establishing the session has to work or else nothing works.  The "get car data" can fail at each stage for reasons that are not clear.  So from testing in the openhab forum it seams better for some cars to get the session and get the data you can and let some channels be null.  Some features like HVAC or Location do not work for some accounts but ther users are happy to get the battery level.  So the binding tries to get what it can and ignore (but log) errors once the session is established.

Reply:
in case httpSession.updateCarData throws an exception, with your current code, you will first change the state to ONLINE and then, to OFFLINE.
I believe `getStatus` should update the thing status only one time in all cases.
If you don't care about exceptions thrown by `httpSession.updateCarData`, maybe do 2 separate try/catch and do not set the thing status to OFFLINE for `updateCarData`.

Reply:
Done.  Thanks for the advice and pointer to an example implementation.

Reply:
Done. 2 stages are clearer now.

Reply:
You forgot to add try/catch around the second stage (updateCarData). Just do nothing in the catch.